src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -545,6 +545,9 @@ public function compile()
             if (!$definition->isPublic()) {
                 $this->privates[$id] = true;
             }
+            if ($definition->isSynthetic()) {
+                $this->synthetics[$id] = true;
```

Comment:
@hhamon: Could you, if you can recall, tell us if there is any reason it was implemented this way in #19146 ?

Maybe it was supposed to hold the generated id instead of a boolean later ? (Thus, it still won't make sense for synthetics service to behaves the same).